src/exchanges/gmocoin-client.js

```diff
@@ -0,0 +1,306 @@
+    try {
+      let remote_id = market.id;
+      let client = null;
+      let key = null;
+      map.keys(k => {
+        if (key.remote_id === remote_id && key.marketObjectType == marketObjectType) {
+          key = k;
+          return;
+        }
+      });
+      if (key === null) {
+        key = { remote_id: remote_id, marketObjectType: marketObjectType };
+      }
+      // construct a client
+      if (!map.has(key)) {
+        let clientArgs = { auth: this.auth, market: market };
+        client = this._createBasicClientThrottled(clientArgs);
+        // we MUST store the promise in here otherwise we will stack up duplicates
+        map.set(key, client);
+      }
+
+      // wait for client to be made!
+      client = await map.get(key);
+      await wait(1000);
+
+      if (marketObjectType === MarketObjectTypes.ticker) {
+        let subscribed = client.subscribeTicker(market);
+        if (subscribed) {
+          client.on("ticker", (ticker, market) => {
+            this.emit("ticker", ticker, market);
+          });
+        }
+      }
+
+      if (marketObjectType === MarketObjectTypes.candle) {
+        let subscribed = client.subscribeCandles(market);
+        if (subscribed) {
+          client.on("candle", (candle, market) => {
+            this.emit("candle", candle, market);
+          });
+        }
+      }
+
+      if (marketObjectType === MarketObjectTypes.trade) {
+        let subscribed = client.subscribeTrades(market);
+        if (subscribed) {
+          client.on("trade", (trade, market) => {
+            this.emit("trade", trade, market);
+          });
+        }
+      }
+
+      if (marketObjectType === MarketObjectTypes.level2update) {
+        let subscribed = client.subscribeLevel2Updates(market);
+        if (subscribed) {
+          client.on("l2update", (l2update, market) => {
+            this.emit("l2update", l2update, market);
+          });
+          client.on("l2snapshot", (l2snapshot, market) => {
+            this.emit("l2snapshot", l2snapshot, market);
+          });
+        }
+      }
+
+      if (marketObjectType === MarketObjectTypes.level2snapshot) {
+        let subscribed = client.subscribeLevel2Snapshots(market);
+        if (subscribed) {
+          client.on("l2snapshot", (l2snapshot, market) => {
+            this.emit("l2snapshot", l2snapshot, market);
+          });
+        }
+      }
+    } catch (ex) {
+      this.emit("error", ex, market);
+    }
+  }
+}
+class GMOCoinSingleClient extends BasicClient {
+  constructor({ wssPath = "wss://api.coin.z.com/ws/public/v1", watcherMs, parent } = {}) {
+    super(wssPath, "GMOCoin", undefined, watcherMs);
+
+    this.hasTickers = true;
+    this.hasTrades = true;
+    this.hasLevel2Snapshots = true;
+    this.parent = parent;
+  }
+
+  _sendPong(id) {
+    this._wss.send(JSON.stringify({ pong: id }));
+  }
+
+  _sendSubTicker(remote_id) {
+    this._wss.send(
+      JSON.stringify({
+        command: "subscribe",
```

Comment:
So according to the documentation you're going to need to add rate limiting.

> Requests from the same IP (subscribe, unsubscribe) are limited to once per second.

You can learn more in the Contributing guide: 
https://github.com/altangent/ccxws/blob/master/CONTRIBUTING.md#rate-limits

Or see a live example in the HitBTC client:
https://github.com/altangent/ccxws/blob/master/src/exchanges/hitbtc-client.js#L22